src/config.py

```diff
@@ -34,6 +64,106 @@ def plugin_keys(cls) -> filter:
         """Return plugin config names in a iterable."""
         return filter(lambda x: x.startswith("plugin_"), cls.keys())
 
+    @validator("durability_synchronous_commit")
+    @classmethod
+    def durability_synchronous_commit_values(cls, value: str) -> Optional[str]:
+        """Check durability_synchronous_commit config option is one of `on`, `remote_apply` or `remote_write`."""
+        if value not in ["on", "remote_apply", "remote_write"]:
```

Comment:
I think we can use [Optional[Literal[val1, val2]]](https://docs.pydantic.dev/1.10/usage/types/#literal-type) instead of writing validators, but that's a nit.

Reply:
Thanks for the suggestion! I created https://github.com/canonical/postgresql-k8s-operator/issues/295 to evaluate that later.